homeassistant/components/config/zwave.py

```diff
@@ -26,13 +27,35 @@ def async_setup(hass):
     hass.http.register_view(ZWaveNodeGroupView)
     hass.http.register_view(ZWaveNodeConfigView)
     hass.http.register_view(ZWaveUserCodeView)
-    hass.http.register_static_path(
-        URL_API_OZW_LOG, hass.config.path(OZW_LOG_FILENAME), False)
+    hass.http.register_view(ZWaveLogView)
     hass.http.register_view(ZWaveConfigWriteView)
 
     return True
 
 
+class ZWaveLogView(HomeAssistantView):
+    """View to read the ZWave log file."""
+
+    url = r"/api/zwave/ozwlog/{lines:\d+}"
+    name = "api:zwave:ozwlog"
+
+    @ha.callback
+    def get(self, request, lines):
+        """Retrieve the lines from ZWave log."""
+        lines = int(lines)
+        hass = request.app['hass']
+        logfilepath = hass.config.path(OZW_LOG_FILENAME)
+
+        with open(logfilepath, 'r') as logfile:
```

Comment:
Oh, you're not allowed to do I/O in the event loop.

Change method signature to `@asyncio.coroutine`

Move the opening of file to a new method on this class and call it `_get_log(self, hass, lines)`.

```python
response = yield from hass.async_add_job(self._get_log, hass, lines)
```

Ignoring pylint for no self use = 👍